scripts/tests/happy_tests.sh

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+
+#
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This file is used to test a single happy test using (almost) the same
+# method as ninja does. Suggest passing $HAPPY_LOG_DIR to set path for logs.
+
+SOURCE="${BASH_SOURCE[0]}"
+SOURCE_DIR="$(cd "$(dirname "$SOURCE")" >/dev/null 2>&1 && pwd)"
+REPO_DIR="$SOURCE_DIR/../../"
+
+export HAPPY_LOG_DIR="${HAPPY_LOG_DIR:-$(mktemp -d)}"
+
+set -x
+env
+
+function happytest_bootstrap() {
+    echo "Bootstrapping Happy Test"
+    set -e
+    # Bootstrap Happy
+    cd "$REPO_DIR/third_party/happy/repo"
+
+    # Override happy log dir config.
+    python3 -c 'import json, os; file = open("happy/conf/main_config.json"); data = json.load(file); data["state_file_prefix"] = os.environ.get("HAPPY_STATE_FILE_PREFIX", data["state_file_prefix"]); data["log_directory"] = os.environ["HAPPY_LOG_DIR"]; data["default_happy_log_dir"] = os.environ["HAPPY_LOG_DIR"]; out = open("happy/conf/main_config.json", "w"); json.dump(data, out);'
```

Comment:
Can you explain why this is required?

If we run this as "root", which it seems we have to, then it writes a root owned filed to the checkout. Please don't do that.

Reply:
In CI, this is used to set the log path of happy test, by default, the logs are written to `/tmp`, and we want to set it to `/tmp/happy_test_logs`, the state file is set to avoid permission issue (can be found at https://github.com/erjiaqing/connectedhomeip/runs/1241935135).

overriting an existing file by root won't change the owner to root.

Reply:
Please don't make edits to checked in files when building or testing. This will dirty everyone's checkout.

Reply:
After a update of happy, how happy tests for chip are using a dedicate `main_conf.json` in chip repo.